discretize/CylMesh.py

```diff
@@ -29,17 +30,39 @@ class CylMesh(BaseTensorMesh, BaseRectangularMesh, InnerProducts, CylView):
 
     _unitDimensions = [1, 2*np.pi, 1]
 
-    def __init__(self, h, x0=None, cartesianOrigin=None):
-        BaseTensorMesh.__init__(self, h, x0)
+    cartesianOrigin = properties.Array(
+        "Cartesian origin of the mesh",
+        dtype=float,
+        shape=('*',)
+    )
+
+    def __init__(self, h=None, x0=None, **kwargs):
+        BaseTensorMesh.__init__(self, h=h, x0=x0, **kwargs)
         assert self.hy.sum() == 2*np.pi, "The 2nd dimension must sum to 2*pi"
         if self.dim == 2:
             print('Warning, a disk mesh has not been tested thoroughly.')
-        cartesianOrigin = (np.zeros(self.dim) if cartesianOrigin is None
-                           else cartesianOrigin)
-        assert len(cartesianOrigin) == self.dim, ("cartesianOrigin must be the "
-                                                  "same length as the dimension"
-                                                  " of the mesh.")
-        self.cartesianOrigin = np.array(cartesianOrigin, dtype=float)
+
+        if 'cartesianOrigin' in kwargs.keys():
+            self.cartesianOrigin = kwargs.pop('cartesianOrigin')
+        else:
+            self.cartesianOrigin = np.zeros(self.dim)
+
+        # assert len(self.cartesianOrigin) == self.dim, (
+        #     "cartesianOrigin must be the same length as the dimension"
+        #     " of the mesh."
+        # )
+        # self.cartesianOrigin = np.array(self.cartesianOrigin, dtype=float)
+
+    @properties.validator('cartesianOrigin')
+    def check_cartesian_origin_shape(self, change):
+        change['value'] = np.array(change['value'], dtype=float).ravel()
```

Comment:
This is cool! I like seeing that you are using the validator to (possibly) change the value.